### PR TITLE
fix(plugin-meetings): refactor getDisplayMedia

### PR DESCRIPTION
**Pull request by abailey-wwt**
Refactor getDisplayMedia to improve logic flow.
Swap video resolution for screen share resolution.

Update of PR: #1816 which incorrectly uses video resolution instead
of the screen share resolution.

Fixes #SPARK-162954

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js [hunk 1]
```diff
@@ -287,28 +287,48 @@ Media.updateTransceiver = ({meetingId, remoteQualityLevel}, peerConnection, tran
...
-            width: resolutionConf.idealWidth || screenResolution.idealWidth
-          }),
-        // Leave last to apply overrides
-        ...resolutionConf
-      };
+  // SDK screen share resolution settings from Webex.init
+  const customResolution = config.screenResolution || {};
+  // user defined share preferences
+  const hasSharePreferences = options.sharePreferences;
+  // eslint-disable-next-line max-len
```
Comment (chrisadubois) [inline packages/node_modules/@webex/plugin-meetings/src/media/index.js:294]: IMO no reason to disable eslint here, it's just as messy to have a comment

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js [hunk 2]
```diff
@@ -287,28 +287,48 @@ Media.updateTransceiver = ({meetingId, remoteQualityLevel}, peerConnection, tran
...
+  const {
+    screenResolution,
+    resolution,
+    videoShareFrameRate,
+    screenFrameRate,
+    aspectRatio
+  } = Config.meetings;
+
+  let shareConstraints = {
+    cursor: 'always',
```
Comment (chrisadubois) [inline packages/node_modules/@webex/plugin-meetings/src/media/index.js:307]: constant>

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js [hunk 3]
```diff
@@ -287,28 +287,48 @@ Media.updateTransceiver = ({meetingId, remoteQualityLevel}, peerConnection, tran
...
-          {
-            frameRate: Config.meetings.screenFrameRate,
-            height: resolutionConf.idealHeight || screenResolution.idealHeight,
-            width: resolutionConf.idealWidth || screenResolution.idealWidth
-          }),
-        // Leave last to apply overrides
-        ...resolutionConf
-      };
+  // SDK screen share resolution settings from Webex.init
+  const customResolution = config.screenResolution || {};
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/media/index.js:291]: nice catch

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js (hunk 2 above)
Comment (adamweeks) [reply packages/node_modules/@webex/plugin-meetings/src/media/index.js:307]: Agreed with @chrisadubois, let's convert this to a constant.

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js (hunk 1 above)
Comment (abailey-wwt) [reply packages/node_modules/@webex/plugin-meetings/src/media/index.js:294]: Cool I didn't know that I could remove this, it was throwing errors about over 80 characters in my IDE. I'll take it out now.

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js (hunk 2 above)
Comment (abailey-wwt) [reply packages/node_modules/@webex/plugin-meetings/src/media/index.js:307]: Good catch, I missed this when I refactored. I not a fan of magic numbers or strings.

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js (hunk 1 above)
Comment (adamweeks) [reply packages/node_modules/@webex/plugin-meetings/src/media/index.js:294]:
```js
  const hasCustomConstraints =
    hasSharePreferences && hasSharePreferences.shareConstraints;
```
Will quiet the eslint warning.

**Comment by adamweeks** [comment]
Test failures are matching those of unmodified meetings plugin code. Merging.

**Pull request merged**